v3/onesignal-metabox/onesignal-metabox.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+defined('ABSPATH') or die('This page may not be accessed directly.');
+
+// Create a meta box
+add_action('add_meta_boxes', 'onesignal_add_metabox');
+
+function onesignal_add_metabox()
+{
+  add_meta_box(
+    'onesignal_metabox', // metabox ID
+    'OneSignal Push Notifications', // title
+    'onesignal_metabox', // callback function
+  );
+}
+
+// Render the meta box
+function onesignal_metabox($post)
+{
+  // View segments API to populate Segments list.
+  $args = array(
+    'headers' => array(
+      'Authorization' => 'Bearer ' . get_option('OneSignalWPSetting')['app_rest_api_key'],
+      'accept' => 'application/json',
+      'content-type' => 'application/json',
+    )
+  );
+
+  // Make the API request, log errors, get segment names.
+  $response = wp_remote_get('https://onesignal.com/api/v1/apps/' . get_option('OneSignalWPSetting')['app_id'] . '/segments', $args);
+  if (is_wp_error($response)) {
+    error_log('API request failed: ' . $response->get_error_message());
+  }
+  $json = json_decode(wp_remote_retrieve_body($response));
+
+  // Meta box content -> js file hides sections depending on whats checked.
+?>
+  <label for="os_update">
+    <input type="checkbox" name="os_update" id="os_update" <?php echo isset($post->os_meta['os_update']) && $post->os_meta['os_update'] == 'on' ? 'checked' : '' ?>>Send notification when post is published</label>
+  <div id="os_options">
+    <label for="os_segment">Send to segment</label>
+    <select name="os_segment" id="os_segment">
+      <option value="All">All</option>
```

Comment:
Seems a little odd to have an "All" option hardcoded when most users will also have a "Total Subscribers" segment or similar. Wdyt about removing this hard-coded option?

Reply:
It's best to have one hardcoded option, in case the segment list isn't returned for any reason.

We have renamed our Default segments over the years, so it's hard to decide on the correct Segment name to choose here.

"All" is what the current plugin uses, so I added this to keep the same behaviour of targeting all subscribed subscriptions.